notes.lua

```diff
@@ -0,0 +1,322 @@
+    local edit_mode = self.note ~= nil
+
+    self:addviews{
+        widgets.Window{
+            frame={w=35,h=20},
+            frame_inset={t=1},
+            autoarrange_subviews=false,
+            subviews={
+                widgets.HotkeyLabel {
+                    key='CUSTOM_ALT_N',
+                    label='Name',
+                    frame={t=0},
+                    on_activate=function() self.subviews.name:setFocus(true) end,
+                },
+                text_editor.TextEditor{
+                    view_id='name',
+                    focus_path='notes/name',
+                    frame={t=1,h=3},
+                    frame_style=gui.FRAME_INTERIOR,
+                    frame_style_b=nil,
+                    init_text=self.note and self.note.name or ''
+                },
+                widgets.HotkeyLabel {
+                    key='CUSTOM_ALT_C',
+                    label='Comment',
+                    frame={t=4},
+                    on_activate=function() self.subviews.comment:setFocus(true) end,
+                },
+                text_editor.TextEditor{
+                    view_id='comment',
+                    frame={t=5,b=3},
+                    focus_path='notes/comment',
+                    frame_style=gui.FRAME_INTERIOR,
+                    init_text=self.note and self.note.comment or ''
+                },
+                widgets.Panel{
+                    view_id='buttons',
+                    frame={b=0,h=3},
+                    autoarrange_subviews=true,
+                    subviews={
+                        edit_mode and widgets.TextButton{
+                            view_id='Save',
+                            frame={h=1},
+                            label='Save',
+                            key='CUSTOM_ALT_S',
+                            on_activate=function() self:saveNote() end,
+                            enabled=function() return #self.subviews.name:getText() > 0 end,
+                        } or widgets.TextButton{
+                            view_id='Create',
+                            frame={h=1},
+                            label='Create',
+                            key='CUSTOM_ALT_S',
+                            on_activate=function() self:createNote() end,
+                            enabled=function() return #self.subviews.name:getText() > 0 end,
+                        },
+                        edit_mode and widgets.TextButton{
+                            view_id='delete',
+                            frame={h=1},
+                            label='Delete',
+                            key='CUSTOM_ALT_D',
+                            on_activate=function() self:deleteNote() end,
+                        } or nil,
+                    }
+                }
+            },
+        },
+    }
+end
+
+function NoteManager:createNote()
+    local x, y, z = pos2xyz(df.global.cursor)
+    if x == nil then
+        print('Enable keyboard cursor to add a note.')
```

Comment:
A mouse-capable way of adding a note is generally expected of current tools. See `gui/blueprint` for a possible example. You click on a button and then click on the map. while the location is being selected, a marker is rendered under the mouse cursor: https://github.com/DFHack/scripts/blob/master/gui/blueprint.lua#L427

Reply:
seems very good, but for now we do not have any `notes` interface where such button can be placed.
Do you think I should replace `notes add` command by mouse cursor selection?
Or this should stay as it is and the new way should appear after there will be a `gui/notes` for managing notes?

Reply:
I think there should be a `gui/notes` entrypoint, but I also think that this dialog should have a way to create (and switch to editing) an additional note

Reply:
This would be confusing. 
1. You clicked on a note to edit it and in a dialog you have option "New note"? Did you current changes to note saved or not? Do it switch to new note overlay mouse hover mode? etc.

2. You creating a new note, and got to moment when you see this dialog. Then you switch to editing. What this really means? You are forced to select a note on the screen now? Its sounds very confusing.
If you want to switch to editing - just select a note on the map. You can do this right now, do not need any additional wat to "switch" to it.
--
I never met such features in my web/desktop experience. Edit dialog is for editing, new dialog is for creating.


I believe the needs you speak about will be fully solved by `gui/notes` tool.
I though I can create it later, but maybe it should be done from the begining to avoid the confusion.